src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiManifest.groovy

```diff
@@ -141,6 +146,44 @@ class JpiManifest extends Manifest {
     }
 
     private static listUpDependencies(Configuration c, boolean optional, StringBuilder buf) {
+        // The category attribute has been introduced in Gradle '5.3'
+        if (GradleVersion.current() >= GradleVersion.version('5.3')) {
+            listUpResolvedDependencies(c, optional, buf)
+        } else {
+            legacyListUpDependencies(c, optional, buf)
+        }
+    }
+
+    private static listUpResolvedDependencies(Configuration c, boolean optional, StringBuilder buf) {
+        def categoryAttribute = Attribute.of(Category.CATEGORY_ATTRIBUTE.name, String)
+
+        c.incoming.resolutionResult.root.dependencies.each { DependencyResult result ->
+            if (result.constraint || !(result instanceof ResolvedDependencyResult)) {
+                return
+            }
+            def selected = ((ResolvedDependencyResult) result).selected
+            // TODO: Find a better way to exclude platform dependencies
+            if (selected.variants.size() == 1 &&
+                    selected.variants.get(0).attributes.getAttribute(categoryAttribute) != Category.LIBRARY) {
```

Comment:
You should be able to use `Category.CATEGORY_ATTRIBUTE` directly (?)

```
selected.variants.get(0).attributes.getAttribute(Category.CATEGORY_ATTRIBUTE).getName() != Category.LIBRARY
```

Reply:
I actually started out with `getAttribute(Category.CATEGORY_ATTRIBUTE)`, but that returns `null`. Is this a bug on the Gradle side?

Reply:
Looks like it is intentional: https://github.com/gradle/gradle/blob/836bd3c13490b66405d3869fec5a95ee3d6fad43/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/metadata/MavenImmutableAttributesFactory.java#L35

Reply:
Uh yes sorry. That would only work during matching. Please leave it as it is for now. Maybe it can be improved as well if we define 'real' jpi variants (see other comment).